src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java

```diff
@@ -220,8 +224,22 @@ public String getKey() {
 			logger.fine("No credential for perforce");
 			return null;
 		}
+
+		// Retrieve item from request
+		StaplerRequest req = Stapler.getCurrentRequest();
+		Job job = req == null ? null : req.findAncestorObject(Job.class);
```

Comment:
This smells wrong. `guessBrowser` is not necessarily called from an HTTP handler thread.